### PR TITLE
feat(emails): Add custom error for users logging in with secondary email

### PR DESCRIPTION
**Pull request by vbudhram**
This PR allows `/account/login` to return a custom error when a user attempts to login with a secondary email. Previously if a user attempted to login with a secondary email, it would return `Account not found error`, which would cause the content-server to prompt to create account 👎 .

Connects with https://github.com/mozilla/fxa-auth-db-mysql/pull/227, which should be merged before this.

### CODE REVIEW

File: lib/routes/account.js [hunk 1]
```diff
@@ -438,6 +438,7 @@ module.exports = (
         checkIsBlockForced()
           .then(() => customs.check(request, email, 'accountLogin'))
           .catch(extractUnblockCode)
+          .then(checkSecondaryEmail)
```
Comment (vladikoff) [inline lib/routes/account.js]: @vbudhram most people will either have an account already or not at all. Does it make sense to move this check below the `readEmailRecord` check? If you find the email via `readEmailRecord `then you don't have to do `checkSecondaryEmail` it seems.

Comment (vbudhram) [inline lib/routes/account.js]: @vladikoff Think your right, the more common case is a user attempting to login with email on accounts table, swapped them around.

**Comment by vbudhram** [comment]
@vladikoff Made updates, mind another r?

File: lib/db.js [hunk 2]
```diff
@@ -856,6 +856,21 @@ module.exports = (
...
 
+  DB.prototype.getSecondaryEmail = function (email) {
+    log.trace({
+      op: 'DB.getSecondaryEmail',
+      email: email
+    })
+
+    return this.pool.get('/email/' + Buffer(email, 'utf8').toString('hex'))
+      .catch((err) => {
+        if(isNotFoundError(err)){
```
Comment (vladikoff) [inline lib/db.js]: `if (` need a space there.

(we should really add this ESLint rule)

File: lib/routes/account.js [hunk 3]
```diff
@@ -489,6 +489,28 @@ module.exports = (
...
+          // know that this can not be used to login.
+          return db.getSecondaryEmail(email)
+            .then((email) => {
+              if (email) {
+                throw error.cannotLoginWithSecondaryEmail()
+              }
+            }, (err) => {
+              // No secondary email exists for this, continue with the regular login flow
+              if (err.errno === error.ERRNO.SECONDARY_EMAIL_UNKNOWN) {
+                return
```
Comment (vladikoff) [inline lib/routes/account.js]: could we add trace logs here

File: lib/routes/account.js [hunk 4]
```diff
@@ -489,6 +489,28 @@ module.exports = (
           throw e
         }
 
+        function checkSecondaryEmail() {
+          if (! features.isSecondaryEmailEnabled()) {
```
Comment (vladikoff) [inline lib/routes/account.js]: probably need a trace log here too before the `if`

File: lib/error.js [hunk 5]
```diff
@@ -619,6 +621,24 @@ AppError.unverifiedPrimaryEmailNewlyCreated = () => {
   })
 }
 
+AppError.cannotLoginWithSecondaryEmail = () => {
+  return new AppError({
+    code: 400,
+    error: 'Bad Request',
+    errno: ERRNO.LOGIN_WITH_SECONDARY_EMAIL,
```
Comment (shane-tomlinson) [inline lib/error.js]: We'll have to add these error numbers to the content server or else [we might end up with strange problems once these surface](https://github.com/mozilla/fxa-content-server/issues/4906).

Comment (vladikoff) [inline lib/error.js]: yeah, that was done

**Pull request merged**